src/app-layout/visual-refresh/trigger-button.tsx

```diff
@@ -33,37 +52,66 @@ function TriggerButton(
     ariaControls,
     onClick,
     testId,
+    disabled = false,
     badge,
     selected = false,
     highContrastHeader,
   }: TriggerButtonProps,
   ref: React.Ref<ButtonProps.Ref>
 ) {
+  const { isMobile } = useAppLayoutInternals();
+
   return (
     <div className={clsx(styles['trigger-wrapper'], !highContrastHeader && styles['remove-high-contrast-header'])}>
-      <button
-        aria-expanded={ariaExpanded}
-        aria-controls={ariaControls}
-        aria-haspopup={true}
-        aria-label={ariaLabel}
-        className={clsx(
-          styles.trigger,
-          {
-            [styles.selected]: selected,
-            [styles.badge]: badge,
-          },
-          className
-        )}
-        onClick={onClick}
-        ref={ref as React.Ref<HTMLButtonElement>}
-        type="button"
-        data-testid={testId}
-      >
-        <span className={clsx(badge && styles['trigger-badge-wrapper'])}>
-          <Icon name={iconName} svg={iconSvg} />
-        </span>
-      </button>
-      {badge && <div className={styles.dot} />}
+      {isMobile ? (
+        <InternalButton
+          ariaExpanded={ariaExpanded}
+          ariaLabel={ariaLabel}
+          ariaControls={ariaControls}
+          className={className}
+          disabled={disabled}
+          ref={ref}
+          formAction="none"
+          iconName={iconName}
+          iconSvg={iconSvg}
+          badge={badge}
+          onClick={onClick}
+          variant="icon"
+          __nativeAttributes={{
+            'aria-haspopup': true,
+            ...(testId && {
+              'data-testid': testId,
+            }),
+          }}
+        />
+      ) : (
+        <>
+          <button
+            aria-expanded={ariaExpanded}
+            aria-controls={ariaControls}
+            aria-haspopup={true}
+            aria-label={ariaLabel}
+            disabled={disabled}
+            className={clsx(
+              styles.trigger,
+              {
+                [styles.selected]: selected,
+                [styles.badge]: badge,
+              },
+              className
+            )}
+            onClick={onClick}
+            ref={ref as Ref<HTMLButtonElement>}
+            type="button"
+            data-testid={testId}
```

Comment:
The data-testid is not set on the mobile version - why is that?

nit: would it make sense to create some commonProps and pass it to each button? That can include ARIA props, onClick, etc. This way it will be easier to see what is actually different.

Reply:
oh, I see it is actually provided as a native prop, but why?

Reply:
it gets added above

```
 __nativeAttributes={{
            'aria-haspopup': true,
            ...(testId && {
              'data-testid': testId,
            }),
          }}
```

Reply:
That is how it is currently done. Not sure as to why.